### PR TITLE
feat: pass parent=self to the WrapperAuthenticator

### PR DESCRIPTION
**Pull request by lahwaacz**
This allows to change the [example configuration](https://github.com/idiap/multiauthenticator?tab=readme-ov-file#configuration) as follows:

```python
from oauthenticator.gitlab import GitLabOAuthenticator
from jupyterhub.auth import PAMAuthenticator

c.GitLabOAuthenticator.client_id: "ZZZZ"
c.GitLabOAuthenticator.client_secret: "AAAAA"
c.GitLabOAuthenticator.oauth_callback_url: "https://jupyterhub.example.com/hub/gitlab/oauth_callback"
c.GitLabOAuthenticator.gitlab_url: "https://gitlab.example.com"

c.MultiAuthenticator.authenticators = [
    (GitLabOAuthenticator, '/gitlab', {}),
    (PAMAuthenticator, "/pam", {"service_name": "PAM"}),
    ...
]

c.JupyterHub.authenticator_class = 'multiauthenticator.multiauthenticator.MultiAuthenticator'
```

### CODE REVIEW

**Review by sgaist — CHANGES_REQUESTED** [review]
Thanks for the feature !

I am using the [conventional commits standard](https://www.conventionalcommits.org/en/v1.0.0/) for the commit messages. Could you rephrase yours following it and add some more information in the commit message ?

**Comment by lahwaacz** [comment]
@sgaist Sure, done.

**Review by sgaist — APPROVED** [review]
